ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosDataAccessor.java

```diff
@@ -144,47 +141,37 @@ void testConnectivity() {
   ResourceResponse<Document> deleteMetadata(CloudBlobMetadata blobMetadata) throws DocumentClientException {
     String docLink = getDocumentLink(blobMetadata.getId());
     RequestOptions options = getRequestOptions(blobMetadata.getPartitionId());
-    options.setPartitionKey(new PartitionKey(blobMetadata.getPartitionId()));
     return executeCosmosAction(() -> asyncDocumentClient.deleteDocument(docLink, options).toBlocking().single(),
         azureMetrics.documentDeleteTime);
   }
 
   /**
-   * Read the blob metadata document in the CosmosDB collection.
-   * @param blobId the {@link BlobId} for which metadata is requested.
-   * @return the {@link ResourceResponse} containing the metadata document.
-   * @throws DocumentClientException if the operation failed.
-   */
-  ResourceResponse<Document> readMetadata(BlobId blobId) throws DocumentClientException {
-    String docLink = getDocumentLink(blobId.getID());
-    RequestOptions options = getRequestOptions(blobId.getPartition().toPathString());
-    return executeCosmosAction(() -> asyncDocumentClient.readDocument(docLink, options).toBlocking().single(),
-        azureMetrics.documentReadTime);
-  }
-
-  /**
-   * Update the blob metadata document in the CosmosDB collection, retrying as necessary.
+   * Update the blob metadata document in the CosmosDB collection.
    * @param blobId the {@link BlobId} for which metadata is replaced.
    * @param fieldName the metadata field to update.
    * @param value the new value for the field.
    * @return the {@link ResourceResponse} returned by the operation, if successful.
    * Returns {@Null} if the document is not found or the field already has the specified value.
-   * @throws DocumentClientException if the operation failed.
+   * @throws DocumentClientException if the record was not found or if the operation failed.
    */
   ResourceResponse<Document> updateMetadata(BlobId blobId, String fieldName, Object value)
       throws DocumentClientException {
-    ResourceResponse<Document> response = readMetadata(blobId);
-    Document doc = response.getResource();
-    if (doc == null) {
-      logger.warn("Blob metadata record not found: {}", blobId.getID());
-      return null;
-    }
+
+    // Read the existing record
+    String docLink = getDocumentLink(blobId.getID());
+    RequestOptions options = getRequestOptions(blobId.getPartition().toPathString());
+    ResourceResponse<Document> readResponse =
+        executeCosmosAction(() -> asyncDocumentClient.readDocument(docLink, options).toBlocking().single(),
+            azureMetrics.documentReadTime);
+    Document doc = readResponse.getResource();
```

Comment:
how is the doc == null case handled now?

Reply:
Per Cosmos it can't happen.  It either returns a document response or throws exception.